service/src/main/java/uk/gov/hmcts/reform/fpl/service/LocalAuthorityUserService.java

```diff
@@ -6,58 +6,60 @@
-    private static final String CASE_TYPE = "CARE_SUPERVISION_EPO";
-
     private final Logger logger = LoggerFactory.getLogger(getClass());
     private final CaseAccessApi caseAccessApi;
+    private final CaseUserApi caseUserApi;
     private final LocalAuthorityUserLookupConfiguration localAuthorityUserLookupConfiguration;
+    private SystemUpdateUserConfiguration userConfig;
     private final AuthTokenGenerator authTokenGenerator;
+    private final IdamClient client;
+    private final Set<String> caseRoles = Set.of("[LASOLICITOR]","[CREATOR]");
 
     @Autowired
     public LocalAuthorityUserService(CaseAccessApi caseAccessApi,
                                      LocalAuthorityUserLookupConfiguration localAuthorityUserLookupConfiguration,
-                                     AuthTokenGenerator authTokenGenerator) {
+                                     AuthTokenGenerator authTokenGenerator,
+                                     CaseUserApi caseUserApi,
+                                     IdamClient idamClient,
+                                     SystemUpdateUserConfiguration userConfig) {
         this.caseAccessApi = caseAccessApi;
         this.localAuthorityUserLookupConfiguration = localAuthorityUserLookupConfiguration;
         this.authTokenGenerator = authTokenGenerator;
+        this.caseUserApi = caseUserApi;
+        this.client = idamClient;
+        this.userConfig = userConfig;
     }
 
-    public void grantUserAccess(String authorization, String creatorUserId, String caseId, String caseLocalAuthority) {
+    public void grantUserAccessWithCaseRole(String caseId, String caseLocalAuthority) {
         findUserIds(caseLocalAuthority).stream()
-            .filter(userId -> !Objects.equals(userId, creatorUserId))
             .forEach(userId -> {
-                logger.debug("Granting user {} access to case {}", userId, caseId);
-
                 try {
-                    caseAccessApi.grantAccessToCase(
-                        authorization,
-                        authTokenGenerator.generate(),
-                        creatorUserId,
-                        JURISDICTION,
-                        CASE_TYPE,
-                        caseId,
-                        new UserId(userId));
-
-                } catch (Exception ex) {
-                    logger.warn("Could not grant user {} access to case {}", userId, caseId, ex);
+                    String authentication = client.authenticateUser(userConfig.getUserName(), userConfig.getPassword());
+                    caseUserApi.updateCaseRolesForUser(authentication, authTokenGenerator.generate(), caseId, userId,
+                        new CaseUser(userId, caseRoles));
+                    logger.info("Added case roles {} to user {}", caseRoles, userId);
+                } catch (Exception exception) {
+                    logger.warn(String.format("Error adding case roles %s to user %s",
+                        caseRoles, userId), exception);
```

Comment:
let's be consistent with how we interpolate strings. {} I think is used everywhere else, no need for String.format method 👍

Reply:
done